src/branch-manager-service/task-service/presubmit-pr.ts

```diff
@@ -13,22 +20,48 @@ import {setStatusOnGithub} from '../shared/github';
 export const handlePresubmitPrTask = functions
   .pubsub
   .topic('presubmit-pr-task')
-  .onPublish(async message => {
-    const {org, repo, pr, branch, sha} = message.json;
+  .onPublish(async (message: functions.pubsub.Message) => {
+    const {org, repo, pr, branches, sha} = message.json as PresubmitPrTaskData;
     const url = `https://presubmit-service-dot-branch-manager.appspot.com/check_pr`;
-    const config = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({org, repo, pr, branch, sha})
-    }
     try {
-      const response = await fetch(url, config).then(response => response.json());
-      await setStatusOnGithub(org, repo, sha, response.status);
+      const results = await Promise.all(branches.map(branch => {
+        const body = JSON.stringify({org, repo, pr, branch, sha});
+        return fetch(url, buildRequestConfig(body)).then(response => response.json());
+      }));      
+      await setStatusOnGithub(org, repo, sha, ...buildGithubStatus(results));
     } catch (e) {
       await setStatusOnGithub(
         org, repo, sha, 'failure', 
-        'The PR does not cleanly cherry-pick onto the targetted branch');
+        'The PR was unable to be checked by the cherry pick checking service');
     }
   });
+
+/** 
+ * Build result information from results from the PR check request, building an array
+ * with the result and description string.
+ */
+function buildGithubStatus(results: PrCheckResult[]): ['failure' | 'success', string] {
+  // Whether all of the PR Checks are passing.
+  let passing;
+  const branchResults = results.map(result => {
+    passing |= result.status === 'success' ? 1 : 0;
```

Comment:
Does this logic work?

```
test = 0;

test |= 1; // Pass
test |= 1; // Pass
test |= 0; // Fail

// Expected test to be "0"

test;
--> 1
```

Reply:
Shouldn't this be `&` and I think you can skip the ternary that transforms into bits.

Reply:
Whoops, your right, I flipped `| and `&` as I was decided if I should track passing or failing

Reply:
I'm just going to push to the fix to master without a PR

Reply:
Sure 👍